src/Uno.UWP/Globalization/DateTimeFormatting/DateTimeFormatter.cs

```diff
@@ -411,97 +356,464 @@ public string Format(DateTimeOffset datetime, string timeZoneId)
+			Append(ToTemplateString(IncludeTimeZone));
 		}
 
-		return result;
-	}
+		return templateBuilder.ToString();
 
-	private static readonly (Regex pattern, string replacement)[] PatternsReplacements =
-		new (string pattern, string replacement)[]
+		void Append(string value)
+		{
+			if (value.Length == 0)
+			{
+				return;
+			}
+
+			if (templateBuilder.Length != 0)
 			{
-				(@"\bMMMM\b", "{month.full}"),
-				(@"\bMMM\b", "{month.abbreviated}"),
-				(@"\bMM\b", "{month.numeric}"),
-				(@"%M\b", "{month.numeric}"),
-				(@"\bM\b", "{month.numeric}"),
-				(@"\bdddd\b", "{dayofweek.full}"),
-				(@"\bddd\b", "{dayofweek.abbreviated}"),
-				(@"\byyyy\b", "{year.full}"),
-				(@"\byy\b", "{year.abbreviated}"),
-				(@"\b(z|zz)\b", "{timezone.abbreviated}"),
-				(@"\byyyy\b", "{year.full}"),
-				(@"\bMMMM\b", "{month.full}"),
-				(@"\bdd\b", "{day.integer(2)}"),
-				(@"%d\b", "{day.integer}"),
-				(@"\bd\b", "{day.integer}"),
-				(@"\bzzz\b", "{timezone.full}"),
-				(@"\bzz\b", "{timezone.abbreviated}"),
-				(@"%z\b", "{timezone}"),
-				(@"\b(HH|hh|H|h)\b", "{hour}"),
-				(@"\b(mm)\b", "{minute.integer(2)}"),
-				(@"\b(m)\b", "{minute.integer}"),
-				(@"\b(mm|m)\b", "{minute}"),
-				(@"\b(ss|s)\b", "{second}"),
-				(@"\btt\b", "{period.abbreviated}"),
+				templateBuilder.Append(' ');
 			}
-			.SelectToArray(x =>
-				(new Regex(x.pattern, RegexOptions.CultureInvariant | RegexOptions.Compiled),
-					x.replacement));
 
-	private IEnumerable<string> BuildPatterns()
+			templateBuilder.Append(value);
+		}
+	}
+
+	private string BuildPattern()
 	{
-		var format = Template;
-		string? sanitizedFormat = default;
+		string datePattern;
+		if (IsLongDate)
+		{
+			datePattern = _firstCulture.DateTimeFormat.LongDatePattern;
+		}
+		else if (IsShortDate)
+		{
+			datePattern = _firstCulture.DateTimeFormat.ShortDatePattern;
+		}
+		else
+		{
+			// NOTE: The original order that was specified in the template string does NOT matter.
+			// That's why all we need to care about is the values of the mentioned properties.
+			// However, we should actually be checking the actual enum value and not just being "None" or not.
+			// But the actual correct implementation that will 100% match WinUI isn't yet clear.
+			// This is a best effort approach.
+			bool hasYear = IncludeYear != YearFormat.None;
+			bool hasMonth = IncludeMonth != MonthFormat.None;
+			bool hasDay = IncludeDay != DayFormat.None;
+			bool hasDayOfWeek = IncludeDayOfWeek != DayOfWeekFormat.None;
+
+			if (hasYear && hasMonth && hasDay && hasDayOfWeek)
+			{
+				datePattern = _firstCulture.DateTimeFormat.LongDatePattern;
+			}
+			else if (hasYear && hasMonth && hasDay)
+			{
+				datePattern = _firstCulture.DateTimeFormat.ShortDatePattern;
+			}
+			else if (hasYear && hasMonth && !hasDayOfWeek)
+			{
+				datePattern = _firstCulture.DateTimeFormat.YearMonthPattern;
+			}
+			else if (hasYear && !hasMonth && !hasDay && !hasDayOfWeek)
+			{
+				datePattern = "yyyy";
```

Comment:
no culture?

Reply:
Here we are just constructing the pattern, which is solely a string. Culture is used when doing the actual formatting.